src/main/resources/assets/src/detailsPage.js

```diff
@@ -9,6 +9,22 @@ export default class Details extends React.Component {
         super(props);
     }
 
+    async cancelAppointment(){
+        this.props.setStateValue('loading', true);
+        // Send the delete request
+        await $.ajax({
+            type: 'DELETE',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/json'
+            },
+            url: '/api/appointments/delete',
```

Comment:
Pass the appointment id to the url: `'/api/appointments/delete/' + this.props.appointment.id`